capstone/src/main/resources/static/user.js

```diff
@@ -39,6 +39,21 @@ function setUpAccountPage(isOrganization, fillForm, data, hide, display) {
   displayNavToggle(hide, display);
 }
 
+/* Add the image form to the profile page */
+function addImageField(formName) {
+  fetch('upload-image').then(response => response.text()).then((data) => {
+    const form = document.getElementById(formName);
+    form.innerHTML = data;
+  });
+
+}
+
+/* upon submission, hide the image form */
+function closeImageForm() {
+  document.getElementById("user-image-form").style.display = "none";
+  setTimeout(function () { window.location.reload(); }, 10000);
```

Comment:
Why did you chose 10s? It seems pretty long.
Also, instead of refreshing the whole page, have you considered just forcing the image to refresh?
You can do it by setting `src` of the image.
Something like this:
```
document.getElementById("public-image-img").setAttribute("src", "get-public-image?email=" + data.email + new Date().getTime());
```

You can even do it multiple times for 10s or something like that.

Reply:
The 10 seconds is arbitrary. On my computer it takes about 8 seconds so I just added a little more time than that to give some leeway.